book/src/action.md

```diff
@@ -1,33 +1,23 @@
-# Action
+# The Action circuit
 
-The action is a mechanism that allows a user to spend or create [notes](notes.md) by proving the right to do so using ZK proofs.
+The Action circuit is a mechanism that ensures that the proposed state transitions follow the Taiga rules. 
+Unlike validity predicates, the Action circuit checks don't depend on the applications involved and are the same for all transactions.
 
-## Spending a note
+As an application state is contained in notes, and a state transition is represented as spending the old notes and creating new ones,
+the Action circuit checks the correctness of the old state by checking properties of the old notes and the correctness of the new state by checking the properties of the new notes.
 
-To spend a note, the users need to prove that they are allowed to do so, namely:
-- To make sure that the correct `AppVP` is checked, prove the relationship between `app_VK` and the application field in the spent note.
-  ![img_3.png](img/action_img_3.png)
-- To make sure the correct `SendVP`(read more about `SendVP` [here](./users.md)) is checked, prove the relationship between `send_VK` and the owner field in the spent note.
-- Compute the nullifier of the spent note (using the nullifier key `nk`) so that it cannot be double-spent. To make sure that the nullifier is computed correctly, prove the relationship between `nk` and the owner field in the spent note.
+### Check the old state (spent notes)
 
-![img_1.png](img/action_img_1.png)
+For spent notes, the Action circuit checks that:
+* the note existed before (can you spend a note that doesn't exist?),
```

Comment:
Oh, I somehow missed this comment!

I feel like it is better define it in a way that we can't spend a note that doesn't exist 😅

I get your point, I think it relates to the same problem of having/not having a local CMtree. Intent notes are dummy anyway, so it is hard to say how much it breaks the description.

But Action circuit does check MT commitment for non-dummy notes, right? The question is if it can make sense to spend a non-dummy intermediate note